test/driver/exception_handling/whole_file_located_error.ml

```diff
@@ -0,0 +1,19 @@
+open Ppxlib
+
+let () =
+  let lexing_loc1 =
+    Lexing.{ pos_fname = "impl.ml"; pos_lnum = 1; pos_bol = 0; pos_cnum = 3 }
+  and lexing_loc2 =
+    Lexing.{ pos_fname = "impl.ml"; pos_lnum = 1; pos_bol = 0; pos_cnum = 7 }
+  in
+  let loc =
+    Location.
+      { loc_start = lexing_loc1; loc_end = lexing_loc2; loc_ghost = false }
+  in
+  Driver.V2.(
+    register_transformation
+      ~impl:(fun _ _ ->
+        Location.raise_errorf ~loc "A located error in a whole file transform")
+      "raise_exc")
```

Comment:
Just as a side-note: if you wanted to make it closer to "reality", you could use the location from the first structure item in your callback, i.e. something along the lines
```suggestion
let () =
  Driver.V2.(
    register_transformation
      ~impl:(fun ctxt str ->
      let loc =
      match str with
      | [] -> Location.in_file (Expansion_context.Base.input_name ctxt)
      | hd :: _ -> hd.pstr_loc
      in
        Location.raise_errorf ~loc "A located error in a whole file transform")
      "raise_exc")
```
(I haven't checked if this would compile). But it's just a side-note; the test is just as good with the hard-coded location you're using.